### ISSUE TITLE
[Bug]: Landscape video playback should still allow rotating to reverse landscape with rotation lock

### ISSUE THREAD

**Issue by intrnl** [mozilla-mobile/fenix#24678]
### Steps to reproduce

1. Open Firefox
2. Enable rotation lock/disable auto rotate
3. Play a landscape video
4. Rotate device between landscape and reverse landscape

### Expected behaviour

The screen should still rotate between the two with rotation lock enabled

### Actual behaviour

It's forced to landscape only, which doesn't match the behavior of other applications (e.g. YouTube, Chromium-based browsers)

### Device name

Poco M3

### Android version

Android 11

### Firefox release type

Firefox Nightly

### Firefox version

101.0a1

**Comment by intrnl** [comment]
While Android does show an indicator if you would like to change your screen orientation, the end result is that the orientation is kept when exiting fullscreen, which causes another hassle.

**Comment by indurs** [comment]
@Mugurell , @kbrosnan can I work on this issue?

**Comment by indurs** [comment]
@Mugurell , to play the video both in landscape and landscape reverse, when auto rotate is disabled we can use 

```
activity.requestedOrientation =
                    ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE

```
instead of 
```
activity.requestedOrientation =
                    ActivityInfo.SCREEN_ORIENTATION_USER_LANDSCAPE
```

in MediaSessionFullscreenFeature
https://github.com/mozilla-mobile/android-components/blob/62fe384048d6c4048c0ddd60befa1158774b3589/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt#L55

I think this is enough to fix this issue, but in order to completely solve the orientation issue in general we need to look further into  MediaSessionFullscreenFeature as you've mentioned. I've explored and discussed about this issue in 
https://github.com/mozilla-mobile/fenix/issues/24040

**Comment by Mugurell** [comment]
Thank you for looking into this @indurs 
I've created https://github.com/mozilla-mobile/android-components/issues/12034 to address the issue based on your findings. Please leave a comment there if you plan to work on it and have that ticket assigned to you.

**Comment by Mugurell** [comment]
Fix landed in Android-components - https://github.com/mozilla-mobile/android-components/pull/12042.
Should be available to test in the next Nightly.

**Comment by intrnl** [comment]
Can confirm that it works as intended now.

```
101.0a1 (Build #2015877587), d4c9c548a+
AC: 101.0.20220429152027, 69f4a3aaa1
GV: 101.0a1-20220429094842
AS: 91.1.2
```

**Comment by SoftVision-LorandJanos** [comment]
Verified as fixed with the latest Nightly 101.0a1 (2022-05-01T17:11:31.722002) build.
Device used: Huawei P30 (Android 10).
Closing the ticket as fixed.
